packages/taquito-local-forging/src/taquito-local-forging.ts

```diff
@@ -22,18 +25,37 @@ export * from './interface';
 export { VERSION } from './version';
 export { ProtocolsHash } from './protocols';
 
-const PROTOCOL_CURRENT = ProtocolsHash.PsQuebecn;
+const PROTOCOL_CURRENT = ProtocolsHash.PsRiotuma;
 
-export function getCodec(codec: CODEC, _proto: ProtocolsHash) {
-  return {
-    encoder: encoders[codec],
-    decoder: (hex: string) => {
-      const consumer = Uint8ArrayConsumer.fromHexString(hex);
-      return decoders[codec](consumer) as any;
-    },
-  };
+export function getCodec(codec: CODEC | CODECPROTO021, _proto: ProtocolsHash) {
+  // use encodersProto021 & decodersProto021 if it's quebec or prior
+  if (_proto === ProtocolsHash.PsQuebecn || ProtoInferiorTo(_proto, ProtocolsHash.PsQuebecn)) {
+    return {
+      encoder: encodersProto021[codec],
+      decoder: (hex: string) => {
+        const consumer = Uint8ArrayConsumer.fromHexString(hex);
+        return decodersProto021[codec](consumer) as any;
+      },
+    };
+  } else {
+    return {
+      encoder: encoders[codec],
+      decoder: (hex: string) => {
+        const consumer = Uint8ArrayConsumer.fromHexString(hex);
+        return decoders[codec](consumer) as any;
+      },
+    };
+  }
+  // TODO: Remove above if else once mainnet migrated into rio protocol and uncommon the return block below
```

Comment:
It looks like this todo needs to be done :)

Reply:
comments addressed will merge this shortly